### PR TITLE
fix pydantic conversion of types that have defaults

### PR DESCRIPTION
**Pull request by thejaminator**
Closes https://github.com/strawberry-graphql/strawberry/issues/2622

Fixes the issue where a pydantic type that is required, with a default, would get converted to a strawberry type that is not required.

This behavior is not consistent with normal strawberry types (see linked issue)

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tests/schema/test_fields.py [hunk 1]
```diff
@@ -147,3 +147,29 @@ def c(self):
...
+    type Query {
+      a: User!
+    }
+    
+    type User {
+      name: String!
+    }
+    """
+
+    assert print_schema(schema) == textwrap.dedent(expected).strip()
```
Comment (thejaminator) [inline tests/schema/test_fields.py:175]: added a test for normal strawberry types to make sure they have this behavior. didn't find a similar existing test like this. though lmk if there already is one.

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to James Chua for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Comment by thejaminator** [comment]
Grear point, will update it.

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+Release type: patch
```
[the remark this answers is not in the document]
Comment (thejaminator) [reply RELEASE.md]: @patrick91 updated :)

**Review by patrick91 — APPROVED** [review]
💯

**Pull request merged**